### PR TITLE
[loader] Add assembly name matching on netcore to fix DefaultContext test

### LINKED ISSUE #15195: [netcore] Make System.Runtime.Loader.Tests.DefaultLoadContextTests.LoadInDefaultContext Pass
The test currently fails with:

```
System.Runtime.Loader.Tests.DefaultLoadContextTests.LoadInDefaultContext [FAIL]
      Assert.Throws() Failure
      Expected: typeof(System.IO.FileNotFoundException)
      Actual:   (No exception was thrown)
      Stack Trace:
           at System.Runtime.Loader.Tests.DefaultLoadContextTests.LoadInDefaultContext()
           at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
```

Determine why and make the test pass.

### PR DESCRIPTION
**Pull request by CoffeeFlux**
Fixes #15195

### CODE REVIEW

**Comment by CoffeeFlux** [comment]
@monojenkins build failed

**Comment by CoffeeFlux** [comment]
@monojenkins build failed

**Comment by CoffeeFlux** [comment]
@monojenkins build failed

File: mono/metadata/assembly.c [hunk 1]
```diff
@@ -2838,7 +2836,7 @@ mono_assembly_binding_applies_to_image (MonoAssemblyLoadContext *alc, MonoImage*
 	MonoAssembly *result_ass = NULL;
 	MonoAssemblyName *result_name = &probed_aname;
 	result_name = mono_assembly_apply_binding (result_name, &dest_name);
-	if (result_name != &probed_aname && !mono_assembly_names_equal (result_name, &probed_aname)) {
+	if (result_name != &probed_aname && !mono_assembly_check_name_match (result_name, &probed_aname)) {
```
Comment (lambdageek) [inline mono/metadata/assembly.c]: Not sure this change is needed.  netcore shouldn't get here (no binding redirections).  I think it should stay an eq check.

File: mono/metadata/metadata.c [hunk 2]
```diff
@@ -516,12 +516,12 @@ static const gint16 tableidx [] = {
 #undef TABLEDEF
 };
 
-/* If TRUE (but also see DISABLE_STICT_STRONG_NAMES #define), Mono will check
+/* On legacy. if TRUE (but also see DISABLE_STICT_STRONG_NAMES #define), Mono will check
```
Comment (lambdageek) [inline mono/metadata/metadata.c]:
```suggestion
/* On legacy. if TRUE (but also see DISABLE_DESKTOP_LOADER #define), Mono will check
```

**Pull request merged**